tensorly/decomposition/_parafac2.py

```diff
@@ -458,15 +536,46 @@ def parafac2(
     non-negative, then :math:`B` will be non-negative, but not the orthogonal `P_i` matrices.
     Consequently, the `B_i` matrices are unlikely to be non-negative.
     """
-    weights, factors, projections = initialize_decomposition(
-        tensor_slices, rank, init=init, svd=svd, random_state=random_state
+
+    (weights, factors, projections) = initialize_decomposition(
+        tensor_slices, rank, init=init, svd=svd, random_state=random_state, mask=mask
     )
     factors = list(factors)
 
+    # Initial missing imputation
+    if mask is not None:
+
+        if init == "random" or init == "svd":
+
+            tensor_slices = _update_imputed(
+                tensor_slices=tensor_slices, mask=mask, method="mode-2"
```

Comment:
We can explicitly pass `decomposition=None` here
```suggestion
                tensor_slices=tensor_slices, mask=mask, decomposition=None, method="mode-2"
```

Reply:
Also addressed in a20b562526de762ff7e1f6296d92136e8348dd5b.